### ISSUE TITLE
VSCode ignores JSDoc's @property when specify @typedef type in 1.43.2 and 1.44.0-insider

### ISSUE THREAD

**Issue by jeffryang24** [microsoft/TypeScript#37714]
Seems like VSCode ignores `@property` when specify type inside `@typedef` in 1.43.2 and 1.44.0-insider. It cause type hint to show not so correct type information and also not detected inside autocomplete/intellisense.

- VSCode Version: 1.43.2 and 1.44.0-insider
- OS Version: Ubuntu 18.04 x86_64

Steps to Reproduce:

1. Create JSDoc type definition file, for example `types.js`.
2. Import the type into `form.js`.
3. Ensure the type hint for the specified JSDoc is working properly (no `any`) and the type is detected available inside autocomplete/intellisense.

Temporary Solution:

- Remove type from `@typedef`, such us from `@typedef {Object<string, any>} Person` into `@typedef Person`.
- Or, remove `<string, any>` from Object, such us `@typedef {Object<string, any>} Person` into `@typedef {Object} Person`.

Reproduce Repo: https://github.com/jeffryang24/sandbox/tree/master/javascript/vscode-jsdoc-type-hint-issue

Does this issue occur when all extensions are disabled?: Yes/~~No~~

**Comment by meszaros-lajos-gyorgy** [comment]
I see something very similar in 1.74.0 (commit: 5235c6bb189b60b01b1f49062f4ffa42384f8c91)

I'm trying to add some extra info using jsdoc to my typescript type:

```ts
/**
 * Color containing red, green, blue and alpha channels
 *
 * @typedef {object} ArxColor
 * @property {number} r - red channel, integer between 0 and 255
 * @property {number} g - green channel, integer between 0 and 255
 * @property {number} b - blue channel, integer between 0 and 255
 * @property {number} a - alpha channel, float between 0.0 and 1.0
 *
 * @see https://github.com/arx/ArxLibertatis/blob/1.2.1/src/graphics/GraphicsFormat.h#L29
 */
export type ArxColor = {
  r: number
  g: number
  b: number
  a: number
}
```

Here's how it looks without typedef:
![image](https://user-images.githubusercontent.com/2386064/207639883-0142ed86-cff9-48ec-a473-4d3a4a660031.png)

And here is how it looks with typedef:
![image](https://user-images.githubusercontent.com/2386064/207640668-103a8b50-b466-4ab6-963c-b71a8190d4c4.png)
